### PR TITLE
Hotfix/change platform reqs

### PR DESCRIPTION
**Pull request by martinyde**
- Change platform requirements
- Add changelog
- Add github actions

This PR is primarily about preparing it for php8.1. In this process some github actions were added and a few fixes to comply with coding standards at severity level 4.

Changing the codesniffer severity config will reveal several issues in this module that should be fixed, but that is for a different day.

This PR depends on https://github.com/itk-dev/pretix-api-client-php/pull/3

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -6,13 +6,16 @@
     "license": "MIT",
     "minimum-stability": "dev",
     "require": {
-        "php": "^7.2",
-        "itk-dev/pretix-api-client-php": "^1.0",
+	    "php": "^8.1",
```
Comment (rimi-itk) [inline composer.json:9]: This line is not indented like other similar lines in this file.

File: composer.json [hunk 2]
```diff
@@ -6,13 +6,16 @@
...
         "nicoeg/dawa": "^1.0"
     },
     "require-dev": {
         "drupal/coder": "^8.3.1",
-        "dealerdirect/phpcodesniffer-composer-installer": "dev-master"
+        "dealerdirect/phpcodesniffer-composer-installer": "^1.0",
+        "phpstan/extension-installer": "^1.2",
+        "mglaman/phpstan-drupal": "^1.1",
+        "phpstan/phpstan-deprecation-rules": "^1.1",
+        "itk-dev/pretix-api-client-php": "^1.0"
```
Comment (rimi-itk) [inline composer.json]: `itk-dev/pretix-api-client-php` must not be a `dev` requirement. The version should be updated to `^1.1` to make sure that we get the PHP 8.1 version.

File: phpstan.neon [hunk 3]
```diff
@@ -0,0 +1,14 @@
+parameters:
+  paths:
+    - src
+    - tests
+  level: 0
+  customRulesetUsed: true
+  ignoreErrors:
+      - '#\Drupal calls should be avoided in classes, use dependency injection instead#'
+      - '#Plugin definitions cannot be altered.#'
```
Comment (rimi-itk) [inline phpstan.neon]: If plugin definitions cannot be altered and we actually do it we may run into problems on runtime.

File: src/Pretix/OrderHelper.php [hunk 4]
```diff
@@ -86,12 +88,16 @@ public function getOrder($organizer, $event, $orderCode) {
...
+                $quota
+            );
         }
         catch (\Exception $exception) {
-          throw $this->clientException('Cannot get quota availability',
-            $exception);
+          throw $this->clientException(
+                'Cannot get quota availability',
+                $exception
+                    );
```
Comment (rimi-itk) [inline src/Pretix/OrderHelper.php]: The coding standards may not check/require this, but please be consistent in how code is indented (cf. lines 69–72).

File: src/Pretix/OrderHelper.php [hunk 5]
```diff
@@ -178,18 +184,24 @@ public function getSubEventAvailabilities(SubEvent $subEvent) {
...
+              $quota
+          );
       }
       catch (\Exception $exception) {
-        throw $this->clientException($this->t('Cannot get quota availability'),
-          $exception);
+        throw $this->clientException(
+              $this->t('Cannot get quota availability'),
+              $exception
+                );
```
Comment (rimi-itk) [inline src/Pretix/OrderHelper.php]: The coding standards may not check/require this, but please be consistent in how code is indented (cf. lines 187–190).

File: src/Pretix/OrderHelper.php [hunk 6]
```diff
@@ -260,22 +275,30 @@ public function ensureWebhook(Client $client) {
...
+              $quota
+          );
       }
       catch (\Exception $exception) {
-        throw $this->clientException('Cannot get quota availability',
-          $exception);
+        throw $this->clientException(
+              'Cannot get quota availability',
+              $exception
+                );
```
Comment (rimi-itk) [inline src/Pretix/OrderHelper.php]: The coding standards may not check/require this, but please be consistent in how code is indented. Your editor should have a “Reformat code” (or similar) function that can help with this.

File: src/ProxyClass/Exporter/CheckInListExporter.php
Comment (rimi-itk) [inline src/ProxyClass/Exporter/CheckInListExporter.php:1]: This auto-generated file should not be updated manually.

File: src/ProxyClass/Exporter/CheckInListPdfExporter.php
Comment (rimi-itk) [inline src/ProxyClass/Exporter/CheckInListPdfExporter.php:1]: This auto-generated file should not be updated manually.

File: src/ProxyClass/Exporter/Manager.php
Comment (rimi-itk) [inline src/ProxyClass/Exporter/Manager.php:1]: This auto-generated file should not be updated manually.

File: src/ProxyClass/Exporter/OrderListExporter.php
Comment (rimi-itk) [inline src/ProxyClass/Exporter/OrderListExporter.php:1]: This auto-generated file should not be updated manually.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (martinyde) [reply composer.json]: Fixed

File: composer.json (hunk 1 above)
Comment (martinyde) [reply composer.json:9]: Fixed

File: src/Pretix/OrderHelper.php (hunk 5 above)
Comment (martinyde) [reply src/Pretix/OrderHelper.php]: Fixed

File: src/Pretix/OrderHelper.php (hunk 6 above)
Comment (martinyde) [reply src/Pretix/OrderHelper.php]: Fixed

File: src/Pretix/OrderHelper.php (hunk 4 above)
Comment (martinyde) [reply src/Pretix/OrderHelper.php]: Fixed

File: src/ProxyClass/Exporter/CheckInListExporter.php
Comment (martinyde) [reply src/ProxyClass/Exporter/CheckInListExporter.php:1]: Removed.

File: src/ProxyClass/Exporter/CheckInListPdfExporter.php
Comment (martinyde) [reply src/ProxyClass/Exporter/CheckInListPdfExporter.php:1]: Removed.

File: src/ProxyClass/Exporter/Manager.php
Comment (martinyde) [reply src/ProxyClass/Exporter/Manager.php:1]: Removed.

File: src/ProxyClass/Exporter/OrderListExporter.php
Comment (martinyde) [reply src/ProxyClass/Exporter/OrderListExporter.php:1]: Removed.

File: phpstan.neon (hunk 3 above)
Comment (martinyde) [reply phpstan.neon]: Line removed

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**